lib/chef/resource/sysctl.rb

```diff
@@ -80,8 +84,13 @@ def coerce_value(v)
 
           directory new_resource.conf_dir
 
+          # construct a string, joining members of new_resource.comment
+          sysctl_lines = Array(new_resource.comment).map { |c| "# #{c.strip}" }
+
+          sysctl_lines << "#{new_resource.key.tr('/', '.')} = #{new_resource.value}\n\n"
```

Comment:
Is there a reason you introduced the tr('/', '.')here?

To keep this no-op for existing users you should just:

sysctl_lines << "#{new_resource.key} = #{new_resource.value}"

That way everyone won't get upgraded resources with the new lines